### DISCUSSION TITLE
Help: unset bindings defined under `:actions`

### DISCUSSION THREAD

**Discussion by citrusmunch** [alphapapa/listen.el#22] [Q&A] [answered]
I'm trying to unset one of the bindings in a `*Listen Queue: ...*` buffer. (I've burned myself by shuffling the queue with `s` by accident and need to manually undo it all, but that can be a separate issue!)

I see it's set under the `:actions` when making the vtable:

https://github.com/alphapapa/listen.el/blob/61c511541af474027c012de7063ddfdb3f8cc780/listen-queue.el#L234

Using `keymap-local-unset` doesn't work of course since it's not in any keymaps. Trying to `keymap-local-set` to nil or a noop lambda doesn't work either. I find it with `C-h b` under ``> `keymap' Property Bindings:`` as a link to the byte code, but I can't seem to find any documentation on that. 

I see your todo note for a transient option so of course this could be easily done eventually. For now, am I overlooking any workarounds? I'm on `GNU Emacs 29.3`.

**Comment by alphapapa** [comment, marked as answer]
You'd have to modify the source code, because those bindings are applied as text properties in the buffer.  The alternative is to specify the actions as a keymap, but that also requires changing the way the commands are defined.  I might do that someday, or patches welcome.

Or, I suppose you could write a function to post-process the queue buffer and manually remove the `s` binding in the text properties, but that would be awkward and error-prone; I wouldn't recommend it.
